Seminars/Sem.01/Pract.01/Martin Gatev/Task08.cpp

```diff
@@ -0,0 +1,150 @@
+#include <iostream>
+
+using std::cin;
+using std::cout;
+
+const int MAX_SIZE_NAME = 31;
+
+struct Student
+{
+	double grade;
+	unsigned int FN;
+	char name[MAX_SIZE_NAME];
+};
+
+struct StudentGroup
+{
+	double averageGrade;
+	int amountOfStudents;
+	Student* groupArr;
+};
+
+void initGroupOfStudents(StudentGroup& group)
+{
+	double sumOfGrades{};
+
+	for (int i = 0; i < group.amountOfStudents; i++)
+	{
+		cin >> group.groupArr->FN >> group.groupArr->name >> group.groupArr->grade;
+
+		sumOfGrades += group.groupArr->grade;
+
+		group.groupArr++;
+	}
+
+	group.averageGrade = sumOfGrades / group.amountOfStudents;
+
+	//returning the pointer to the start
+	for (int i = 0; i < group.amountOfStudents; i++)
+	{
+		group.groupArr--;
+	}
+
```

Comment:
You could have used cin>>group.groupArr[i] , this way you will not change the pointer